File: example_workflows/demo_account_setup_TS/test_create_log_score.ts
/**
 * Test script for creating log scores - complete end-to-end workflow.
 */

import "dotenv/config";
import { createLog } from "./basic_logging.js";
import { createLLMEvaluator } from "./create_evaluator.js";
import { createLogScore } from "./create_log_scores.js";

const TEST_MODEL = process.env.TEST_MODEL || "gpt-4o";

async function sleep(ms: number): Promise<void> {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

async function main() {
  console.log("Test Create Log Score - Complete Workflow\n");

  // Step 1: Create a log
  console.log("[1] Creating log...");
  const logData = await createLog({
    model: TEST_MODEL,
    inputMessages: [{ role: "user", content: "What is machine learning?" }],
    outputMessage: {
      role: "assistant",
      content: "Machine learning is a subset of AI that enables systems to learn from experience.",
    },
    customIdentifier: `test_score_log_${Date.now()}`,
  });
  const logId = logData.unique_id;

  // Step 2: Create an evaluator
  console.log("\n[2] Creating evaluator...");
  const evaluatorData = await createLLMEvaluator({
    name: "Test Response Quality Evaluator",
    evaluatorSlug: `test_response_quality_${Date.now()}`,
    evaluatorDefinition:
      "Evaluate response quality based on accuracy.\n" +
      "<llm_input>{{input}}</llm_input>\n<llm_output>{{output}}</llm_output>",
    scoringRubric: "0.0=Poor, 0.5=Average, 1.0=Excellent",
  });
  const evaluatorSlug = evaluatorData.evaluator_slug;
  const evaluatorId = evaluatorData.id;

  // Step 3: Wait for processing
  console.log("\n[3] Waiting 2s for processing...");
  await sleep(2000);

  // Step 4: Create score
  console.log("\n[4] Creating score...");
  if (!logId || !evaluatorId) {
    throw new Error("Log ID or Evaluator ID is missing");
  }

  await createLogScore({
    logId,
    evaluatorId,
    evaluatorSlug,
    score: 0.85,
    reasoning: "The response accurately explains machine learning.",
  });

  console.log("\nTest completed successfully!");
}

main().catch(console.error);

File: platform/demo-setup-python/test_create_log_score.py
#!/usr/bin/env python3
"""Test script for creating log scores."""

from basic_logging import create_log
from create_evaluator import create_llm_evaluator
from create_log_scores import create_log_score
from dotenv import load_dotenv
import os
import time

load_dotenv(override=True)

# Test configuration from environment variables
TEST_MODEL = os.getenv("TEST_MODEL", "gpt-4o")
TEST_EVALUATOR_TEMPERATURE = float(os.getenv("TEST_EVALUATOR_TEMPERATURE", "0.1"))
TEST_EVALUATOR_MAX_TOKENS = int(os.getenv("TEST_EVALUATOR_MAX_TOKENS", "200"))

print('=' * 80)
print('Testing Create Log Score - Complete Workflow')
print('=' * 80)

# Step 1: Create a log
print('\n📝 Step 1: Creating a log...')
log_data = create_log(
    model=TEST_MODEL,
    input_messages=[{'role': 'user', 'content': 'What is machine learning?'}],
    output_message={
        'role': 'assistant', 
        'content': 'Machine learning is a subset of artificial intelligence that enables systems to learn and improve from experience without being explicitly programmed.'
    },
    custom_identifier=f'test_score_log_{int(time.time())}'
)
log_id = log_data.get('unique_id')
print(f'✓ Log created with ID: {log_id}')

# Step 2: Create an evaluator
print('\n📊 Step 2: Creating an evaluator...')
evaluator_data = create_llm_evaluator(
    name='Test Response Quality Evaluator',
    evaluator_slug=f'test_response_quality_{int(time.time())}',
    evaluator_definition=(
        'Evaluate the response quality based on accuracy and completeness.\n'
        '<llm_input>{{input}}</llm_input>\n'
        '<llm_output>{{output}}</llm_output>'
    ),
    scoring_rubric='0.0=Poor, 0.5=Average, 1.0=Excellent',
    description='Test evaluator for log scoring',
    model_options={
        "temperature": TEST_EVALUATOR_TEMPERATURE,
        "max_tokens": TEST_EVALUATOR_MAX_TOKENS
    }
)
evaluator_slug = evaluator_data.get('evaluator_slug')
print(f'✓ Evaluator created with slug: {evaluator_slug}')

# Step 3: Wait a moment
print('\n⏳ Waiting 2 seconds for processing...')
time.sleep(2)

# Step 4: Create a score on the log
print('\n⭐ Step 3: Creating score on the log...')
try:
    score_data = create_log_score(
        log_id=log_id,
        evaluator_slug=evaluator_slug,
        score=0.85,
        reasoning='The response accurately explains machine learning and provides a clear, complete definition.'
    )
    print(f'\n✅ SUCCESS! Score created!')
    print(f'Score ID: {score_data.get("id", "N/A")}')
    # Show the correct value field
    if score_data.get('numerical_value') is not None:
        print(f'Score Value (numerical): {score_data.get("numerical_value")}')
    elif score_data.get('string_value'):
        print(f'Score Value (string): {score_data.get("string_value")}')
    elif score_data.get('boolean_value') is not None:
        print(f'Score Value (boolean): {score_data.get("boolean_value")}')
    print(f'\n{"="*80}')
    print('Test completed successfully!')
    print(f'{"="*80}')
except Exception as e:
    print(f'\n❌ Error creating score: {e}')
    import traceback
    traceback.print_exc()
